nav2_amcl/src/amcl_node.cpp

```diff
@@ -624,134 +594,39 @@ AmclNode::laserReceived(sensor_msgs::msg::LaserScan::ConstSharedPtr laser_scan)
   if (!getOdomPose(latest_odom_pose_, pose.v[0], pose.v[1], pose.v[2],
     laser_scan->header.stamp, base_frame_id_))
   {
-    RCLCPP_DEBUG(get_logger(), "Couldn't determine robot's pose associated with laser scan");
+    RCLCPP_ERROR(get_logger(), "Couldn't determine robot's pose associated with laser scan");
     return;
   }
 
-
   pf_vector_t delta = pf_vector_zero();
-
-  if (pf_init_) {
-    // Compute change in pose
-    // delta = pf_vector_coord_sub(pose, pf_odom_pose_);
-    delta.v[0] = pose.v[0] - pf_odom_pose_.v[0];
-    delta.v[1] = pose.v[1] - pf_odom_pose_.v[1];
-    delta.v[2] = angleutils::angle_diff(pose.v[2], pf_odom_pose_.v[2]);
-
-    // See if we should update the filter
-    bool update = fabs(delta.v[0]) > d_thresh_ ||
-      fabs(delta.v[1]) > d_thresh_ ||
-      fabs(delta.v[2]) > a_thresh_;
-    update = update || m_force_update;
-    m_force_update = false;
-
-    // Set the laser update flags
-    if (update) {
-      for (unsigned int i = 0; i < lasers_update_.size(); i++) {
-        lasers_update_[i] = true;
-      }
-    }
-  }
-
   bool force_publication = false;
   if (!pf_init_) {
-    // Pose at last filter update
```

Comment:
I'd leave comments here where possible. AMCL is a pretty considerably complex thing to try to learn reading code without some hints.